api/services/email_mailgun.js

```diff
@@ -79,24 +82,36 @@ module.exports = {
       return callback(undefined, 'Email sent')
     }
 
-    var addresses = R.map(R.objOf('address'), R.keys(data.email))
+    var shapeEmailBody = recipient =>
+      recipient.content.slice(1).join('\n\n')
 
-    var emailArray = R.zipWith(R.merge)(addresses)(R.values(data.email))
+    var grabSubjects = recipient =>
+      recipient.content[0]
 
-    var sendEmail = (recipient, cb) => {
-      var address = recipient.address
-      var emailBody = recipient.content.slice(1).join('\n\n')
-      var subject = recipient.content[0]
-      mg.sendText('messenger@friendsch.org', address, subject, emailBody, error =>
-        error ? cb({error: error.toString(), recipient: recipient.address}, null) : cb(null, address)
+    var toKey = R.compose(R.map(R.objOf('to')), R.keys)
+    var textKey = R.compose(R.map(R.objOf('text')), R.values, R.map(shapeEmailBody))
+    var subjectKey = R.compose(R.map(R.objOf('subject')), R.values, R.map(grabSubjects))
+
+    var emailsData = R.converge(R.zipWith(R.merge), [R.converge(R.zipWith(R.merge), [toKey, textKey]), subjectKey])(data.email)
+    var emails = R.map(R.assoc('from', 'messenger@friendsch.org'), emailsData)
+
+    var sendEmail = (email, cb) => {
```

Comment:
If we define this as `email => cb` we could change `asyncArray` to be defined as `emails.map(sendEmail)`